fractions/swagger/src/main/java/org/wildfly/swarm/swagger/runtime/SwaggerArchivePreparer.java

```diff
@@ -73,10 +89,25 @@ public void process() {
         if (this.deploymentContext != null && this.deploymentContext.isImplicit()) {
             return;
         }
+
+        // Append the JAX-RS application path to the context path (as the default value).
+        // This value can always be overridden by the various settings in place.
+        final String restApplicationPath = getRestApplicationPath();
+        if (!restApplicationPath.isEmpty()) {
+            String path = contextPath.get() + restApplicationPath;
+            path = path.replaceAll("//", "/");
+            if (!Objects.equals(contextPath.get(), path)) {
+                contextPath.set(path);
+            }
+        }
+
         if (archive.getName().endsWith(".war")) {
             // Create a JAX-RS deployment archive
             WARArchive deployment = archive.as(WARArchive.class);
             deployment.addModule("io.swagger");
+            deployment.addAsServiceProvider(Extension.class, SwaggerExtension.class);
```

Comment:
You don't need this line and the META-INF/services file for the Extension. One or the other will suffice.

Does this actually work? I think `SwaggerExtension` needs to be in `.deployment` package so it's available within user deployment.

Reply:
Ah! That might have been the reason it did not get picked up. I had to add it explicitly for it to work. I will run another test after the 1471 PR has been approved.

Reply:
@kenfinnigan How does the `.deployment` package work? Is it added **automatically** to the archive? Do I need to add it as a module (with slot=deployment)?

I am trying to modify this PR and running in to issue. The [create-a-fraction](https://github.com/wildfly-swarm/wildfly-swarm/blob/master/docs/howto/create-a-fraction/index.adoc) doc does not provide enough clarity on this behavior. What I did learn is that classes defined in `.deployment` are not available to `.runtime`. Are there any samples that I can see to get some inspiration?

Reply:
It is not added automatically, but a jboss-module module with the slot of `:deployment` is created (and is excluded from the normal API or runtime class-space). A fraction can then do a `@DeploymentModule` to add the `:deployment` module to a deployment. Not necessarily very clear or straight-forward, but that's how it works. :)

Reply:
@bobmcwhirter Thank you for the explanation. It makes a lot more sense now. Is there a way to define an module as an optional dependency in the `module.conf` file? This solve relies on the availability of CDI. But having CDI is not mandatory. So, can I define an optional dependency on `org.wildfly.swarm.cdi` in the `module.conf` file so that I don't run in to linkage issues?

Reply:
Or rather .. put another way. If I define a dependency in `module.conf`, does that mean that dependency is exported to the archive? Or are they unrelated?

Reply:
Yep, just prefix the line with a literal `*`.

e.g.

https://github.com/wildfly-swarm/wildfly-swarm/blob/master/fractions/jolokia/module.conf#L4

Jolokia optionally has a dependency on keycloak, if it's around.

Reply:
No, it will *not* be exported. It'll just be available to things within the `:runtime` and such modules so they themselves can link against it.

If you want a module exported (or imported? depending on how you look at it) into the deployment, then once against just `@DeploymentModule` it, which will add it to the deployment. But alas `@DeploymentModule` doesn't support optionals.

You *could* add a deployment processor, which should be able to `addModule` on the deployment archive, which ultimately allows full control over adding modules, as it's just scribbling the `jboss-deployment-structure.xml`, which is what `@DeploymentModule` does also, ultimately.

https://github.com/wildfly-swarm/wildfly-swarm/blob/master/core/spi/src/main/java/org/wildfly/swarm/spi/api/JBossDeploymentStructureContainer.java#L39

https://github.com/wildfly-swarm/wildfly-swarm/blob/master/core/spi/src/main/java/org/wildfly/swarm/spi/api/Module.java